instrumentation/jdbc/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jdbc/test/JdbcInstrumentationTest.java

```diff
@@ -0,0 +1,1365 @@
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.testing.assertj.TraceAssert;
+import io.opentelemetry.semconv.ServerAttributes;
+import io.opentelemetry.semconv.incubating.CodeIncubatingAttributes;
+import io.opentelemetry.semconv.incubating.DbIncubatingAttributes;
+import java.beans.PropertyVetoException;
+import java.io.Closeable;
+import java.io.IOException;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import javax.sql.DataSource;
+import org.apache.derby.jdbc.EmbeddedDataSource;
+import org.apache.derby.jdbc.EmbeddedDriver;
+import org.h2.jdbcx.JdbcDataSource;
+import org.hsqldb.jdbc.JDBCDriver;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class JdbcInstrumentationTest {
+
+  @RegisterExtension static final AutoCleanupExtension cleanup = AutoCleanupExtension.create();
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static String dbName;
+  private static String dbNameLower;
+  private static Map<String, String> jdbcUrls;
+  private static Map<String, String> jdbcDriverClassNames;
+  private static Map<String, String> jdbcUserNames;
+  private static Properties connectionProps;
+  // JDBC Connection pool name (i.e. HikariCP) -> Map<dbName, Datasource>
+  private static Map<String, Map<String, DataSource>> cpDatasources;
+
+  @BeforeAll
+  static void setUp() {
+    dbName = "jdbcUnitTest";
+    dbNameLower = dbName.toLowerCase(Locale.ROOT);
+    jdbcUrls =
+        ImmutableMap.of(
+            "h2", "jdbc:h2:mem:" + dbName,
+            "derby", "jdbc:derby:memory:" + dbName,
+            "hsqldb", "jdbc:hsqldb:mem:" + dbName);
+
+    jdbcDriverClassNames =
+        ImmutableMap.of(
+            "h2", "org.h2.Driver",
+            "derby", "org.apache.derby.jdbc.EmbeddedDriver",
+            "hsqldb", "org.hsqldb.jdbc.JDBCDriver");
+
+    jdbcUserNames = new HashMap<>();
+    jdbcUserNames.put("derby", "APP");
+    jdbcUserNames.put("h2", null);
+    jdbcUserNames.put("hsqldb", "SA");
```

Comment:
you could have initialized these in the field declaration as the groovy test did